src/pip/_vendor/pygments/lexer.py

```diff
@@ -207,7 +207,7 @@ def _preprocess_lexer_input(self, text):
                 text, _ = guess_decode(text)
             elif self.encoding == 'chardet':
                 try:
-                    from pip._vendor import chardet
+                    import chardet
```

Comment:
Can this use chardet-normalizer too? Also why is the CI OK with this change not being present in the patch…?

Reply:
Forgot to mention it. `pygments` is used by `rich`, which doesn't pass `encoding="chardet"` (at least when I grep for `chardet` in `rich` I get no results), so this code is not reached by pip.

I added a commit to patch it out